src/qibolab/instruments/port.py

```diff
@@ -0,0 +1,96 @@
+from qibo.config import raise_error
+
+
+class Port:
+    """Abstract interface for instrument parameters.
+
+    These parameters are exposed to the user through :class:`qibolab.channels.Channel`.
+
+    Drivers should subclass this interface and implement the getters
+    and setters for all the parameters that are available for the
+    corresponding instruments.
+
+    Each port is identified by the ``name`` attribute.
+    Note that the type of the identifier can be different of each port implementation.
+    """
+
+    # NOTE: We can convert all these to ``@abstractmethod`` but then
+    # each driver would have to implement all of them an raise errors
+    # for parameters that are not supported.
+
+    def __init__(self, name):
+        self.name = name
+
+    @property
+    def offset(self):
+        """DC offset that is applied to this port."""
+        raise_error(NotImplementedError, "Instruments do not support offset.")
+
+    @offset.setter
+    def offset(self, value):
+        raise_error(NotImplementedError, "Instruments do not support offset.")
```

Comment:
From the discussion about attributes & properties: maybe it would be worth to make all of these attributes, since it will reduce the boilerplate and would allow us to better use both properties and attributes in subclasses

See https://github.com/qiboteam/qibolab/pull/469#discussion_r1229510134